cmd/mesh/rpc_handler.go

```diff
@@ -86,8 +110,19 @@ func (handler *rpcHandler) AddOrders(signedOrdersRaw []*json.RawMessage, opts rp
 }
 
 // AddPeer is called when an RPC client calls AddPeer,
-func (handler *rpcHandler) AddPeer(peerInfo peerstore.PeerInfo) error {
+func (handler *rpcHandler) AddPeer(peerInfo peerstore.PeerInfo) (errRes error) {
 	log.Debug("received AddPeer request via RPC")
+	// Catch panics, log stack trace and return RPC error message
+	defer func() {
+		if err := recover(); err != nil {
```

Comment:
[`recover()` returns an `interface{}` and is not guaranteed to return an `error`](https://golang.org/pkg/builtin/#recover). Typically I would write this as:

```go
if r := recover(); r != nil {
    var ok bool
    var internalErr error
    if internalErr, ok = r.(error); !ok {
        // If r is not of type error, convert it.
        internalErr = fmt.Errorf("Recovered from non-error: (%T) %v", e, e)
    }
    // Log internalErr here...
}
```